### ISSUE TITLE
Improve remote detection

### ISSUE THREAD

**Issue by michael-k** [atom/open-on-github#48]
**tl;dr:** If no remote detected and no remote named origin, but only one github remote exists: take this one.

I have to branches, `master` and `feature`.  Both exist on github and locally, but *only* master has a section in `.git/config`:
```
[branch "master"]
	remote = organization
	merge = refs/heads/master
```
(Note: The remote is not named `origin`.)

When I call `Open on Github: File` on branch `master`, everything works fine.  When I do the same on `feature` I get: `No URL defined for remote: null`.

I expected *open-on-github* to simply use the only existing remote (or if there are more remotes, but only one github remote, to use the github remote) for `Open on Github`.

When I rename the remote to `origin`, everything works as expected.

**Comment by atd** [comment]
I have another use case regarding remote detection

With the following `.git/config`

```
[remote "origin"]
        url = git@organization-host:pear2pear.git
        fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
        remote = origin
        merge = refs/heads/master
[remote "github"]
        url = git@github.com:P2Pvalue/pear2pear.git
        fetch = +refs/heads/*:refs/remotes/github/*
```
*open-on-github* opens `https://www.organization-host.com/pear2pear/...`

**Comment by gangsarswapurba** [comment]
it works for me! 

thanks 👍
